lib/ansible/template/native_helpers.py

```diff
@@ -28,6 +34,33 @@ def visit_Name(self, node):
         return ast.Constant(value=_JSON_MAP[node.id])
 
 
+def _is_unsafe(value: t.Any) -> bool:
+    """
+    Our helper function, which will also recursively check dict and
+    list entries due to the fact that they may be repr'd and contain
+    a key or value which contains jinja2 syntax and would otherwise
+    lose the AnsibleUnsafe value.
+    """
+    to_check = [value]
+
+    while True:
+        if not to_check:
+            break
+
+        val = to_check.pop(0)
+
+        if isinstance(val, AnsibleUndefined):
+            continue
+        if isinstance(val, Mapping):
+            to_check.extend(val.values())
```

Comment:
Dunno if we want to also check keys here or not- DT is a lot more strict about "keys can't be templated" in most cases, but devel is ... not so much.

Reply:
Probably a good idea to check, the current code doesn't do it but that shouldn't mean this should not check in any case. I can see something returning `{{ hostvars[inventory_hostname]['my_var'].keys() | first }}` or something along those lines.